src/plugins/usage_collection/server/plugin.ts

```diff
@@ -49,8 +49,26 @@ export class UsageCollectionPlugin implements Plugin<CollectorSet> {
       maximumWaitTimeForAllCollectorsInS: config.maximumWaitTimeForAllCollectorsInS,
     });
 
+    const globalConfig = await this.initializerContext.config.legacy.globalConfig$
+      .pipe(first())
+      .toPromise();
+
     const router = core.http.createRouter();
-    setupRoutes(router, () => this.savedObjects);
+    setupRoutes({
+      router,
+      getSavedObjects: () => this.savedObjects,
+      collectorSet,
+      config: {
+        allowAnonymous: core.status.isStatusPageAnonymous(),
+        kibanaIndex: globalConfig.kibana.index,
+        kibanaVersion: this.initializerContext.env.packageInfo.version,
+        server: core.http.getServerInfo(),
+        uuid: this.initializerContext.env.instanceUuid,
+      },
+      metrics: core.metrics,
+      overallStatus$: core.status.overall$,
+      getAuthState: core.http.auth.get,
```

Comment:
Note: this API is deprecated on CoreSetup in favor of the CoreStart equivalent. However that makes this very awkward to consume since the route must be registered during setup.

It seems the best API would `KibanaRequest.auth.get` however I think we're a bit limited by how KibanaRequest is constructed. It would need access to some of HttpServer's internals (specifically, the AuthStateStorage class). Another option would be to expose this on `RequestHandlerContext` but that seems awkward IMO to have request details split across two different APIs.

@restrry have you thought through the best way to handle this? I couldn't find any relevant issue.

Reply:
> However that makes this very awkward to consume since the route must be registered during setup.

Unfortunately, this is a common problem for all the `start contract` API. I'd keep this as is until we come up with a better option than `getStartServices`. I remember this has been discussed on a weekly call, but cannot find an appropriate issue to reconsider lifecycles.

Reply:
I was able to remove this completely from this PR, so no need right now 👍